### ISSUE TITLE
Recieve error when install node-sass by yarn . 

### ISSUE THREAD

**Issue by NicolasAndersen** [sass/node-sass#1781]
I have installed node-sass by yarn, but I get error: 
```
error /home/zendkofy/public_html/react-redux/visualsliderdesigner/node_modules/node-sass: Command failed.
Exit code: 1
Command: sh
Arguments: -c node scripts/install.js
Directory: /home/zendkofy/public_html/react-redux/visualsliderdesigner/node_modules/node-sass
Output:
path.js:7
    throw new TypeError('Path must be a string. Received ' + inspect(path));
    ^

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.join (path.js:1211:7)
    at /home/zendkofy/public_html/react-redux/visualsliderdesigner/node_modules/node-sass/scripts/install.js:122:26
    at /home/zendkofy/public_html/react-redux/visualsliderdesigner/node_modules/mkdirp/index.js:30:20
    at FSReqWrap.oncomplete (fs.js:123:15)
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```
Could you please let me know to fix this error ?

**Comment by xzyfer** [comment]
@nschonni this also looks tangentially related to the cache patch. My guess is yarn isn't BC with npm config.

**Comment by xzyfer** [comment]
@NicolasAndersen what version of yarn are you using?

**Comment by scottcarlson** [comment]
@xzyfer I'm having the same issue on yarn `v0.16.1`

Modifying the yarn.lock file back to node-sass version `3.10.0` is the temporary workaround I'm using right now.

**Comment by xzyfer** [comment]
Thanks @scottcarlson this is a bug in yarn that was exposed by a recent change in node-sass.

**Comment by xzyfer** [comment]
Opened an issue with yarn - https://github.com/yarnpkg/yarn/issues/1632

**Comment by NicolasAndersen** [comment]
thank @xzyfer and @scottcarlson . I using yarn version 0.16.1. I have removed folder node_modules then use `npm install` and it worked. Thank you again. You can close this ticket.

**Comment by xzyfer** [comment]
I'll leave this open. This is an issue for yarn support.

**Comment by daltones** [comment]
I just solved it by setting the `cache` config to yarn. It seems like yarn does set all `npm_config_*` env vars, but `cache` doesn't exist by default.

`$ yarn config set cache ~/.my-yarn-cache-dir`

**Comment by xzyfer** [comment]
Interesting observation @daltones. That's a case we currently do no handle gracefully.

**Comment by xzyfer** [comment]
I have a work around in #1782.

**Comment by jbruni** [comment]
Thanks for the workaround, @daltones. This worked for me:

```
jbruni@vm:~$ yarn cache dir
yarn cache v0.16.1
/home/jbruni/.yarn-cache
Done in 0.05s.
jbruni@vm:~$ yarn config set cache ~/.yarn-cache
yarn config v0.16.1
success Set "cache" to "/home/jbruni/.yarn-cache".
Done in 0.10s.
```

**`yarn cache dir`** returns the current cache dir.

**Comment by xzyfer** [comment]
I've released v3.11.1 with a patch until yarn is fixed.

**Comment by magixus** [comment]
this command helped me a lot 
`yarn --cache-folder ~/.my-yarn-cache-dir/`
